src/thor/bidirectional_astar.cc

```diff
@@ -237,20 +236,22 @@ inline bool BidirectionalAStar::ExpandForwardInner(GraphReader& graphreader,
     return true; // This is an edge we _could_ have expanded, so return true
   }
 
-  // TODO: actually use time_info
-  const uint64_t localtime = 0;
-  const uint32_t tz_index = 0;
+  // Skip this edge if no access is allowed (based on costing method)
+  // or if a complex restriction prevents transition onto this edge.
+  // if its not time dependent set to 0 for Allowed and Restricted methods below
+  const uint64_t localtime = time_info.valid ? time_info.local_time : 0;
   int restriction_idx = -1;
-  if (!costing_->Allowed(meta.edge, pred, tile, meta.edge_id, localtime, tz_index, restriction_idx) ||
+  if (!costing_->Allowed(meta.edge, pred, tile, meta.edge_id, localtime, time_info.timezone_index,
+                         restriction_idx) ||
       costing_->Restricted(meta.edge, pred, edgelabels_forward_, tile, meta.edge_id, true,
-                           &edgestatus_forward_, localtime, tz_index)) {
+                           &edgestatus_forward_, localtime, time_info.timezone_index)) {
     return false;
   }
 
   // Get cost. Separate out transition cost.
   Cost transition_cost = costing_->TransitionCost(meta.edge, nodeinfo, pred);
-  // TODO: actually use time_info
-  Cost newcost = pred.cost() + transition_cost + costing_->EdgeCost(meta.edge, tile);
+  Cost newcost =
+      pred.cost() + transition_cost + costing_->EdgeCost(meta.edge, tile, time_info.second_of_week);
```

Comment:
actually use the time of the week to let the costing pick the right historical data to use